packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-one-transformer.test.ts

```diff
@@ -466,3 +466,123 @@ test('recursive @hasOne relationships are supported if DataStore is enabled', ()
   const schema = parse(out.schema);
   validateModelSchema(schema);
 });
+
+describe('Pre Processing Has One Tests', () => {
+  let transformer: GraphQLTransform;
+  const hasGeneratedField = (doc: DocumentNode, objectType: string, fieldName: string): boolean => {
+    let hasField = false;
+    doc?.definitions?.forEach(def => {
+      if ((def.kind === 'ObjectTypeDefinition' || def.kind === 'ObjectTypeExtension') && def.name.value === objectType) {
+        def?.fields?.forEach(field => {
+          if (field.name.value === fieldName) {
+            hasField = true;
+          }
+        });
+      }
+    });
+    return hasField;
+  };
+
+  const hasGeneratedFieldArgument = (doc: DocumentNode, objectType: string, fieldName: string, generatedFieldName: string): boolean => {
+    let hasFieldArgument = false;
+    doc?.definitions?.forEach(def => {
+      if ((def.kind === 'ObjectTypeDefinition' || def.kind === 'ObjectTypeExtension') && def.name.value === objectType) {
+        def?.fields?.forEach(field => {
+          if (field.name.value === fieldName) {
+            field?.directives?.forEach(dir => {
+              if (dir.name.value === 'hasOne') {
+                dir?.arguments?.forEach(arg => {
+                  if (arg.name.value === 'fields') {
+                    if (arg.value.kind === 'ListValue' && arg.value.values[0].kind === 'StringValue' && arg.value.values[0].value === generatedFieldName) {
+                      hasFieldArgument = true;
+                    }
+                    else if (arg.value.kind === 'StringValue' && arg.value.value === generatedFieldName) {
+                      hasFieldArgument = true;
+                    }
+                  }
+                });
+              }
+            });
+          }
```

Comment:
Nit: Is it possible to reduce the nesting here using `filter` and `some`?
May be like `doc.definitions.filter(def => (def.kind === '' and so on) && def.fields.some(f => f.name.value === fieldName))`.